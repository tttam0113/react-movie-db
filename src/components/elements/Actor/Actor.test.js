import React from 'react';
import { shallow } from 'enzyme';
import Actor from './Actor';

const actor = {
    character: 'Charactor Name',
    name: 'Actor Name',
    profile_path: '/feD4iQF7dug3oZDLYZgXBscJ1gM.jpg'
};

it('should render Actor correctly', () => {
    const wrapper = shallow(<Actor actor={actor}/>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('img').length).toBe(1);
    expect(wrapper.find('.rmdb-actor-name').text()).toEqual(actor.name);
    expect(wrapper.find('.rmdb-actor-character').text()).toEqual(actor.character);
});
