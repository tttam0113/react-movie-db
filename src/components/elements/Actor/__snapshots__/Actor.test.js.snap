// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Actor correctly 1`] = `
<div
  className="rmdb-actor"
>
  <img
    alt="actorthumb"
    src="https://image.tmdb.org/t/p/w154/feD4iQF7dug3oZDLYZgXBscJ1gM.jpg"
  />
  <span
    className="rmdb-actor-name"
  >
    Actor Name
  </span>
  <span
    className="rmdb-actor-character"
  >
    Charactor Name
  </span>
</div>
`;
